lib/MetaCPAN/Middleware/Static.pm

```diff
@@ -0,0 +1,106 @@
+package MetaCPAN::Middleware::Static;
+use strict;
+use warnings;
+use Plack::Builder;
+use Plack::App::File;
+use Cwd qw(cwd);
+
+sub new { bless {}, $_[0] }
+
+my $hour_ttl = 60 * 60;
+my $day_ttl  = $hour_ttl * 24;
+my $year_ttl = $day_ttl * 365;
+
+sub wrap {
+    my ( $self, $app, %args ) = @_;
+    my $root_dir = $args{root} || cwd;
+    my $dev_mode = $args{dev_mode}
+        || ( $ENV{PLACK_ENV} && $ENV{PLACK_ENV} eq 'development' );
+    my $tempdir = $args{temp_dir};
+
+    my @js_files = map {"/static/js/$_.js"} (
+        qw(
+            jquery.min
+            jquery.tablesorter
+            jquery.relatize_date
+            jquery.qtip.min
+            jquery.autocomplete.min
+            mousetrap.min
+            shCore
+            shBrushPerl
+            shBrushPlain
+            shBrushYaml
+            shBrushJScript
+            shBrushDiff
+            shBrushCpp
+            shBrushCPANChanges
+            cpan
+            toolbar
+            github
+            contributors
+            dropdown
+            bootstrap/bootstrap-dropdown
+            bootstrap/bootstrap-collapse
+            bootstrap/bootstrap-modal
+            bootstrap/bootstrap-tooltip
+            bootstrap/bootstrap-affix
+            bootstrap-slidepanel
+            syntaxhighlighter
+            ),
+    );
+
+    my @css_files = map { my $f = $_; $f =~ s{^\Q$root_dir\E/root/}{/}; $f }
+        glob "$root_dir/root/static/css/*.css";
+
+    my @less_files = ('/static/less/style.less');
+
+    builder {
+        if ( !$dev_mode ) {
+            enable 'Assets::FileCached' => (
+                files => [ map "root$_", @js_files ],
+
+                # need js compressor
+                #filter => sub { },
```

Comment:
I'll push a commit adding JavaScript::Minifier::XS, which is what was used before.  I'd like to start using closure compiler, but that requires more things to be installed.